File: Ackermann-function/Java/ackermann-function-1.java
import java.math.BigInteger;

class AckermannFunction{
    public static BigInteger ack(BigInteger m, BigInteger n) {
        return m.equals(BigInteger.ZERO)
                ? n.add(BigInteger.ONE)
                : ack(m.subtract(BigInteger.ONE),
                            n.equals(BigInteger.ZERO) ? BigInteger.ONE : ack(m, n.subtract(BigInteger.ONE)));
    }

    public static void main(String[] args){
        for(long m = 0; m < 4; m++){
            for(long n = 0; n < 14; n++){
                System.out.println(ack(BigInteger.valueOf(m), BigInteger.valueOf(n)));
            }
        }
    }
}

File: Ackermann-function/Ruby/ackermann-function-1.rb
def ack(m, n)
  if m == 0
    n + 1
  elsif n == 0
    ack(m-1, 1)
  else
    ack(m-1, ack(m, n-1))
  end
end

for m in 0..3 do
    for n in 0..13 do
        puts ack(m,n)
    end
end
